modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java

```diff
@@ -173,7 +173,7 @@ public AbstractKotlinCodegen() {
         typeMapping.put("DateTime", "java.time.LocalDateTime");
 
         instantiationTypes.put("array", "kotlin.arrayOf");
-        instantiationTypes.put("list", "kotlin.arrayOf");
+        instantiationTypes.put("list", "kotlin.listOf");
```

Comment:
This will also impact other Kotline generators (e.g. client).

@jimschubert @dr4ke616, are you ok with that?

Reply:
Array was being used over List because of the serializer. That's since been changed, so this shouldn't be an issue as far as language support is concerned. We could always just generate the client to hit a service created by the Springboot generator to verify.

However, this will break client consumers who have coded to array.

Another thing to consider is that Kotlin arrays are compiled to Java arrays. These are more lightweight than List. So this change may also have an impact on resource limited consumers (Android).

Reply:
@jimschubert Ah ok, I'll change it back in that case. Out of interest though is list being used as a type in OpenApi specs? I thought this map was from openapi types but I don't see list anywhere

Reply:
@herojan these types represent openapi data types as well as other common types that have been normalized in `DefaultCodegen`. I'm on mobile, so it's hard to search GitHub, but if I remember correctly the DefaultCodegen logic (originally targeting mainly Java) treated OpenAPI array types as array of they were primitives or list if not. It's been about a year since I looked through that code, so that's pretty off memory. But, for example, you could have a type `File` and instantiate it with some custom infrastructure type like `MyFileMaker`.

Reply:
@jimschubert Ah right that makes sense, good to know. Thanks!